carbonmark/components/ProjectFilterModal/index.tsx

```diff
@@ -85,69 +92,100 @@ export const ProjectFilterModal: FC<ProjectFilterModalProps> = (props) => {
     value: vintage,
   }));
 
+  const fetchProjects = (values: ModalFieldValues) => {
+    const { search, sort } = router.query;
+    const query = search ? { ...values, search, sort } : { ...values, sort };
+    router.replace({ query }, undefined, { shallow: true });
+  };
+
   const onSubmit = (values: ModalFieldValues) => {
-    const { search } = router.query;
-    // Maintain any search value
-    const query = search ? { search, ...values } : values;
-    router.replace(
-      { query },
-      undefined,
-      { shallow: true } // don't refetch props nor reload page
-    );
-    //Close the modal on submit
-    props.onToggleModal?.();
+    fetchProjects(values);
+    if (!isValidating) {
+      props.onToggleModal?.();
+    }
+  };
```

Comment:
Not sure this is necessary since the watch is fetching anyway on change.. Maybe we should remove the Apply button all together..